### PR TITLE
Create unload tests for dev panel.

### PR DESCRIPTION
**Pull request by Gozala**

### CODE REVIEW

**Comment by Gozala** [comment]
@erikvold can you please take a look at this ?

File: test/test-dev-panel.js [hunk 1]
```diff
@@ -316,4 +316,97 @@ exports["test createView panel"] = test(function*(assert) {
...
+  // try catch since we don't care if it's exception or different
+  // panel and devtools may start throwing in a future.
+  try {
+    const toolbox = yield openToolbox(MyPanel);
+    const otherPanel = yield getCurrentPanel(toolbox);
+
+    assert.ok(!(otherPanel instanceof MyPanel),
+              "requested panel was not open as it was unregistered");
+  } catch (error) {
+    assert.pass("failed to open toolbox with panel that was unregistered");
```
Comment (erikvold) [inline test/test-dev-panel.js]: we should evaluate the error message here to make sure it is what we expect it to be.

File: test/test-dev-panel.js [hunk 2]
```diff
@@ -316,4 +316,97 @@ exports["test createView panel"] = test(function*(assert) {
...
+  // At the moment devtools won't throw if requested panel isn't
+  // present and will just open different one, but we still use
+  // try catch since we don't care if it's exception or different
+  // panel and devtools may start throwing in a future.
+  try {
+    const toolbox = yield openToolbox(MyPanel);
+    const otherPanel = yield getCurrentPanel(toolbox);
+
+    assert.ok(!(otherPanel instanceof MyPanel),
+              "requested panel was not open as it was unregistered");
```
Comment (erikvold) [inline test/test-dev-panel.js]: shouldn't we `assert.fail()` if it gets this far?

File: test/test-dev-panel.js [hunk 3]
```diff
@@ -316,4 +316,97 @@ exports["test createView panel"] = test(function*(assert) {
...
+            "current panel is not instance of MyPanel");
+
+  yield closeToolbox();
+
+  // At the moment devtools won't throw if requested panel isn't
+  // present and will just open different one, but we still use
+  // try catch since we don't care if it's exception or different
+  // panel and devtools may start throwing in a future.
+  try {
+    const toolbox = yield openToolbox(MyPanel);
```
Comment (erikvold) [inline test/test-dev-panel.js]: this should not throw right? if so can we move it out the the `try/catch`?

File: test/test-dev-panel.js [hunk 4]
```diff
@@ -316,4 +316,97 @@ exports["test createView panel"] = test(function*(assert) {
...
```
Comment (erikvold) [inline test/test-dev-panel.js]: same as above.

File: test/test-dev-panel.js [hunk 5]
```diff
@@ -316,4 +316,97 @@ exports["test createView panel"] = test(function*(assert) {
...
+    const otherPanel = yield getCurrentPanel(toolbox);
+
+    assert.ok(!(otherPanel instanceof MyPanel),
+              "requested panel was not open as it was unregistered");
+  } catch (error) {
+    assert.pass("failed to open toolbox with panel that was unregistered");
```
Comment (erikvold) [inline test/test-dev-panel.js]: same as above

**Comment by andymckay** [comment]
Closing due to inactivity.

**Pull request closed without merging**